src/foam/util/SafetyUtil.java

```diff
@@ -0,0 +1,23 @@
+package foam.util;
+
+
+import java.math.BigDecimal;
+
+public class SafetyUtil{
+
+  public static int compare(Object o1 , Object o2) {
+    
+    if (o1 == null && o2 == null) {
```

Comment:
Replace this if statement with these three:

if ( o1 == null && o2 == null ) Return 0;
if ( o2 == null ) return 1;
if ( o1 == null ) return -1;

See why?